contracts/carbon/Strategies.sol

```diff
@@ -635,7 +638,16 @@ abstract contract Strategies is Initializable {
                 payable(address(owner)).sendValue(txValue - depositAmount);
             }
         } else if (depositAmount > 0) {
-            token.safeTransferFrom(owner, address(this), depositAmount);
+            if (validateDepositAmount) {
+                uint256 prevBalance = token.balanceOf(address(this));
+                token.safeTransferFrom(owner, address(this), depositAmount);
+                uint256 newBalance = token.balanceOf(address(this));
+                if (newBalance - prevBalance != depositAmount) {
+                    revert BalanceMismatch();
```

Comment:
Minor: the error-message, IMO, should be 'TransferFeeNotAllowed' or something like that.

Reply:
Please discuss with Yudi to reach an agreement on that - his preference is BalanceMismatch, for me it also makes more sense.

Reply:
there's no to know the cause is a transfer fee